## Урок 3. Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных

#### 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [5]:
from math import sqrt
data = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]
n = len(data)
mean = sum(data)/n
Var_population = sum([(i - mean)**2 for i in data])/n
Var_sample = sum([(i - mean)**2 for i in data])/(n-1)
std_population = sqrt(Var_population)
std_sample = sqrt(Var_sample)
mean, Var_population, Var_sample, std_population, std_sample

(65.3, 950.11, 1000.1157894736842, 30.823854398825596, 31.624607341019814)

##### Ответ: <br>среднее арифметическое = 65.3 <br> дисперсия для ген. совокупности = 950.11  <br> дисперсия для выборки ~ 1000.12  <br>ско для ген. совокупности ~ 30.82  <br> ско для выборки ~ 31.62  <br>

#### 2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. <br>Какова вероятность того, что 3 мяча белые?

In [15]:
# p1: ящик1 - 0 белых, ящик2 - 3 белых
p1 = (3/8) * (2/7) * (5/12) * (4/11) * (3/10) * (7/9) * 4
# p2: ящик1 - 1 белый, ящик2 - 2 белых
p2 = 2 * (5/8) * (3/7) * 6 * (5/12) * (4/11) * (7/10) * (6/9)
# p3: ящик1 - 2 белых, ящик2 - 1 белый
p3 = (5/8) * (4/7) * 4 * (5/12) * (7/11) * (6/10) * (5/9)
p = p1 + p2 + p3
p

0.36868686868686873

##### Ответ: <br>вероятность вытащить 3 белых мяча: ~ 36.87 %

#### 3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

In [16]:
precision_sum = 0.9 + 0.8 + 0.6
p1 = 0.9 / precision_sum
p2 = 0.8 / precision_sum
p3 = 0.6 / precision_sum
p1, p2, p3

(0.3913043478260869, 0.34782608695652173, 0.26086956521739124)

##### Ответ: <br>вероятность, что выстрел произвел первый спортсмен ~ 39.13 % <br> второй спортсмен ~ 34.78 % <br> третий спортсмен ~ 26.09 %  <br>

#### 4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе.<br> Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. <br>Какова вероятность, что он учится: <ol><li>на факультете A?</li><li>на факультете B?</li><li>на факультете C?</li></ol>

In [17]:
weighted_sum = 0.25*0.8 +  0.25*0.7 + 0.5*0.9
p1 = (0.25*0.8) / weighted_sum
p2 = (0.25*0.7) / weighted_sum
p3 = (0.5*0.9) / weighted_sum
p1, p2, p3

(0.24242424242424246, 0.21212121212121213, 0.5454545454545455)

##### Ответ: <br>вероятность, что студент учится на факультете А ~ 24.24 % <br> факультете B ~ 21.21 % <br> факультете C ~ 54.55 %  <br>

#### 5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. <br>Какова вероятность того, что в первый месяц выйдут из строя: <ol><li>все детали?</li><li>только две детали?</li><li>хотя бы одна деталь?</li><li>от одной до двух деталей?</li></ol>

In [28]:
all_details = 0.1*0.2*0.25
two_details = 0.1*0.2*0.75 + 0.1*0.25*0.8 + 0.25*0.2*0.9
at_least_one = 1 - 0.9*0.8*0.75
one_to_two_details = at_least_one - all_details
all_details, two_details, at_least_one, one_to_two_details

(0.005000000000000001,
 0.08000000000000002,
 0.45999999999999996,
 0.45499999999999996)

##### Ответ: <br>вероятность, что из строя выйдут все детали ~ 0.5 % <br> 2 детали ~ 8.00 % <br> хотя бы 1 деталь ~ 46.00 %  <br>